### PR TITLE
Update Bismark wrapper to 0.19.1 - and some other edits

### PR DESCRIPTION
**Pull request by scholtalbers**
Updated the wrapper to use the latest Bismark version and use the new bam type 'unsorted' so galaxy doesn't complain about the inability to set the metadata.

### CODE REVIEW

File: tools/bismark/bismark_mapping/bismark_bowtie2_wrapper.xml [hunk 1]
```diff
@@ -1,28 +1,17 @@
...
-        <exit_code range="1:" />
-        <exit_code range=":-1" />
-        <regex match="Error:" />
-        <regex match="Exception:" />
-    </stdio>
     <command interpreter="python">
-<![CDATA[
-        bismark_wrapper.py
+        <![CDATA[
+        $__tool_directory__/bismark_wrapper.py
```
Comment (bgruening) [inline tools/bismark/bismark_mapping/bismark_bowtie2_wrapper.xml]: can you please single quote those paths?

File: tools/bismark/bismark_methyl_extractor/bismark_methylation_extractor.xml [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
+        <requirement type="package" version="2.3.4.2">bowtie2</requirement>
     </requirements>
 
-    <parallelism method="basic"></parallelism>
-
     <command interpreter="python">
-<![CDATA[
-        bismark_methylation_extractor.py
+        <![CDATA[
+        $__tool_directory__/bismark_methylation_extractor.py
```
Comment (bgruening) [inline tools/bismark/bismark_methyl_extractor/bismark_methylation_extractor.xml]: This one please as well.

File: tools/bismark/bismark_pretty_report/bismark2report_wrapper.xml [hunk 3]
```diff
@@ -1,25 +1,16 @@
-<tool id="bismark_pretty_report" name="Bismark Pretty Report" version="0.16.3">
+<tool id="bismark_pretty_report" name="Bismark Pretty Report" version="0.19.1" profile="17.01">
```
Comment (bgruening) [inline tools/bismark/bismark_pretty_report/bismark2report_wrapper.xml]: 18.01?

File: tools/bismark/bismark_pretty_report/bismark2report_wrapper.xml [hunk 4]
```diff
@@ -1,25 +1,16 @@
...
-        <regex match="Exception:" />
-    </stdio>
-
     <command interpreter="python">
-<![CDATA[
-        bismark2report_wrapper.py
-
-        --tool_dir "$__tool_directory__"
+        <![CDATA[
+        $__tool_directory__/bismark2report_wrapper.py
```
Comment (bgruening) [inline tools/bismark/bismark_pretty_report/bismark2report_wrapper.xml]: single quote

File: tools/bismark/bismark_pretty_report/bismark2report_wrapper.xml [hunk 5]
```diff
@@ -36,40 +27,49 @@
             --nucleotide_report "$additional_reports['nucleotide']"
         #end if
 
-        ##--output_html_report $output_html_report
-        --output_html_report_link "$output_html_report_link"
-        --output_dir "$output_html_report_link.files_path"
+        --output_html_report $output_html_report
```
Comment (bgruening) [inline tools/bismark/bismark_pretty_report/bismark2report_wrapper.xml]: single quote for all filepath

File: tools/bismark/bismark_pretty_report/bismark2report_wrapper.xml [hunk 6]
```diff
@@ -36,40 +27,49 @@
...
 
-        ##--log_report $log_report
+        #if $separate_logfile:
+            --log_report $log_report
```
Comment (bgruening) [inline tools/bismark/bismark_pretty_report/bismark2report_wrapper.xml]: single quote

File: tools/bismark/tool_dependencies.xml [hunk 7]
```diff
@@ -1,41 +0,0 @@
-<?xml version="1.0"?>
```
Comment (bgruening) [inline tools/bismark/tool_dependencies.xml]: Oh yes, lets get rid of this one!!!

**Comment by bgruening** [comment]
Super cool! Is this still WIP as indicated in the title?

**Comment by scholtalbers** [comment]
yes, the tests still need some more work

File: tools/bismark/bismark_pretty_report/bismark2report_wrapper.xml (hunk 3 above)
Comment (scholtalbers) [reply tools/bismark/bismark_pretty_report/bismark2report_wrapper.xml]: this particular one doesn't need 18.01 (no bam in/out) but 17.01 for the updated stdio error checks

**Comment by scholtalbers** [comment]
@bgruening ready for review ;)

**Pull request merged**